scripts/publish-packages.sh

```diff
@@ -0,0 +1,25 @@
+SCRIPTS_DIR="$(dirname "$0")"
+cd "${SCRIPTS_DIR}/../packages"
+for d in */ ; do
+    cd $d
+
+    PACKAGE_NAME=$(node -p "require('./package.json').name")
+    PACKAGE_VERSION=$(node -p -e "require('./package.json').version")
+    echo "Will check if version ${PACKAGE_VERSION} of ${PACKAGE_NAME} should be published."
+
+    PUBLISHED_VERSION=$(npm view "${PACKAGE_NAME}" version)
+    echo "Current version of ${PACKAGE_NAME} in repository is ${PUBLISHED_VERSION}."
+
+    if [[ ${PUBLISHED_VERSION} != ${PACKAGE_VERSION} ]]; then
```

Comment:
I wonder if this is a problem for pre-version-1.0.0 version stamps... according to the semantic versioning spec, any 0.x.y versions are non-orderable and non-deterministic as "0.1.1-alpha" or "alpha-0123"

Reply:
`npm view` (and other commands) don't sort versions. If no version is specified, they return the version specified by the "latest" tag.
https://docs.npmjs.com/cli/v9/commands/npm-view#description

As long as we publish versions in increasing order, I don't think this would be a problem.

Reply:
But alternatively, instead of viewing the latest version, this could try viewing the version specified in package.json to see if it exists. In that case, it might be easier to process the results of `npm view --json ...`.

Reply:
I thought of looking if the version in package.json exists in any of the published versions. But really, if we try to publish a version that already exists it isn't a big deal… we just get an error that the version already exists. So I think the simpler check about the last version is sufficient.

Reply:
a safe  bet would be to up any package versions in question to 1.0.0 which will behave as expected going forward for any "latest" commands....

Reply:
@msilva-broad I am not understanding your concern. What I have will return the "latest" tag, which will always be the last thing published.

Reply:
for a given package, if the artifact store only has pre-version 1 version stamps, they are non-sortable and can't reliably answer "latest" version.

"0.0.1"
"0.0.2"
"alpha-0.1"

all are considered unsortable and unable to determine "latest" according to the sem-ver spec.

Reply:
It can reliably answer which is the "latest" version, because that is determined by a "dist-tag", not by sorting versions.
> {"name":"@terra-ui-packages/components","dist-tags":{"latest":"0.0.2"}, ...
https://us-central1-npm.pkg.dev/dsp-artifact-registry/terra-ui-packages/@terra-ui-packages/components

Reply:
All I’m trying to say is that if the version of the library is not what the current version is, try publishing it. If you have screwed up and specified a version that already exists (but is not the current), the publish will fail because it doesn’t let you override existing versions. That seems ok to to me because you need to correct the version in that case to something that isn’t published.

Reply:
sorry for the latency on my end... meetings.  I think this is fine as is, and we can iterate later if we need to.